### PR TITLE
Feature/list cell size

### PR DESCRIPTION
**Pull request by SilverGull**

### CODE REVIEW

File: src/kirby/components/avatar/avatar.component.tns.ts [hunk 1]
```diff
@@ -63,5 +66,6 @@ export class AvatarComponent implements OnInit {
       }
     }, 100);
   }
-
 }
+
+registerElement(AVATAR_COMPONENT_SELECTOR, () => require('./avatar.component').AvatarComponent);
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/avatar/avatar.component.tns.ts]: Out of curiosity, this is done to support native code in the component or?

File: src/kirby/components/list/list-cell-line/list-cell-line.component.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.primary {
+  font-size: 1.2em;
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list-cell-line/list-cell-line.component.scss]: Should we not use the defines sizes in the variable classes?

```scss
$sizes: (
  xxxl:         70,    // 4.375rem,         
  xxl:          60,    // 3.75rem,          
  xl:           50,    // 3.125rem,       
  l:            32,    // 2rem,           
  m:            16,    // 1rem,  - Same as gutter
  s:            12,    // .75rem,         
  xs:           8,     // .5rem, - Same as gutter-small
  xxs:          4,     // .25rem,     
  gutter:       16,    // 1rem,  - helper
  gutter-small: 8      // 0.5rem - helper
);
```

File: src/kirby/components/list/list-cell-line/list-cell-line.component.tns.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Label {
+  font-size: 13;
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list-cell-line/list-cell-line.component.tns.scss]: Should we not use the defines sizes in the variable classes?

```scss
$sizes: (
  xxxl:         70,    // 4.375rem,         
  xxl:          60,    // 3.75rem,          
  xl:           50,    // 3.125rem,       
  l:            32,    // 2rem,           
  m:            16,    // 1rem,  - Same as gutter
  s:            12,    // .75rem,         
  xs:           8,     // .5rem, - Same as gutter-small
  xxs:          4,     // .25rem,     
  gutter:       16,    // 1rem,  - helper
  gutter-small: 8      // 0.5rem - helper
);
```

File: src/kirby/components/list/list-cell/list-cell.component.ts [hunk 4]
```diff
@@ -1,26 +1,103 @@
...
+  private readonly horisontalAlignmentDefault: horisontalAlignment = 'left';
+  private readonly verticalAlignmentDefault: verticalAlignment = 'center';
+  private readonly widthDefault = 1;
+
+  @Input() horisontalAlignment: horisontalAlignment = this.horisontalAlignmentDefault;
+  @Input() verticalAlignment: verticalAlignment = this.verticalAlignmentDefault;
+  @Input() width = this.widthDefault;
 
-  @HostBinding('style.flex-grow') flexGrowHost: string;
+  @HostBinding('style.flex-basis')
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list-cell/list-cell.component.ts]: Are we following the Angular style guide in Kirby?
If yes we should not have `_` prefixed private variables [style-03-04](https://angular.io/guide/styleguide#style-03-04).

File: src/kirby/components/list/list-cell/list-cell.component.ts [hunk 5]
```diff
@@ -1,26 +1,103 @@
...
+  }
+
+  private setStyle() {
+    this._flexBasisHost = this.getWidth();
+    this._flexAlignHost = this.getAlignItems();
+    this._flexJustifyHost = this.getJustifyContent();
   }
 
+  getWidth(): string {
+    if (
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list-cell/list-cell.component.ts]: This is my personal style, but i find it hard to read this if statement. What is wrong with:

```javascript
 if (this.width && this.width > 0) {
   return `${this.width * 100}%`;
 }
```

Or is it just prettier acting up here?

File: src/kirby/components/list/list-cell-line/list-cell-line.component.tns.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  @Input()
+  primary: boolean;
+
+  constructor() {
+    super();
+    this.primary = false;
+  }
+}
+
+registerElement(LIST_CELL_LINE_COMPONENT_SELECTOR, () => require('./list-cell-line.component').ListCellLineComponent);
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list-cell-line/list-cell-line.component.tns.ts]: This is my limited nativescript speaking. But do we really need to almost identical components beside this line?

File: src/kirby/components/list/list-cell/list-cell.component.ts [hunk 7]
```diff
@@ -1,26 +1,103 @@
...
   styleUrls: ['./list-cell.component.scss']
 })
 export class ListCellComponent implements OnInit {
-  @Input() lineOne: string | number;
-  @Input() lineTwo: string | number;
-  @Input() lineThree: string | number;
-  @Input() alignHorisontal: flexAlign = 'flex-start';
-  @Input() alignVertical: flexAlign = 'flex-start';
-  @Input() flexGrow: string | number;
+  private readonly horisontalAlignmentDefault: horisontalAlignment = 'left';
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list-cell/list-cell.component.ts]: If we follow the official angular style guide, place fix the members sequence in [style-05-14](https://angular.io/guide/styleguide#style-05-14)

This also goes for you functions.

File: tsconfig.json [hunk 8]
```diff
@@ -20,6 +20,10 @@
         "paths": {
             "~/*": [
                 "src/*"
+            ],
```
Comment (jonasjensen-brolstark) [inline tsconfig.json]: Was this not removed explicit in an other PR. Or do i remember wrong?

File: src/app/examples/examples.common.ts [hunk 9]
```diff
@@ -7,11 +7,11 @@ import { ChartExampleComponent } from './chart-example/chart-example.component';
...
 
 export const COMPONENT_DECLARATIONS: any[] = [
-  ButtonExampleComponent,
-  CardExampleComponent,
-  ListExampleComponent,
-  GridExampleComponent,
-  AvatarExampleComponent,
-  ChartExampleComponent,
-  FontsExampleComponent,
+    ButtonExampleComponent,
```
Comment (jonasjensen-brolstark) [inline src/app/examples/examples.common.ts]: Looks like you have you tab space set to 4 and not 2. 

I am unsure what is default for the Kirby project?

File: src/kirby/components/list/list-cell/list-cell.component.ts (hunk 5 above)
Comment (SilverGull) [reply src/kirby/components/list/list-cell/list-cell.component.ts]: must have been prettier. A reformat fixed it :-)

File: src/kirby/components/list/list-cell-line/list-cell-line.component.scss (hunk 2 above)
Comment (SilverGull) [reply src/kirby/components/list/list-cell-line/list-cell-line.component.scss]: We surely should. Now using $font-size() and $font-weigth()

File: src/kirby/components/list/list-cell-line/list-cell-line.component.tns.scss (hunk 3 above)
Comment (SilverGull) [reply src/kirby/components/list/list-cell-line/list-cell-line.component.tns.scss]: We surely should. Now using $font-size() and $font-weigth()

File: src/kirby/components/list/list-cell/list-cell.component.ts (hunk 4 above)
Comment (SilverGull) [reply src/kirby/components/list/list-cell/list-cell.component.ts]: saw that one as well :-) fixed

File: src/kirby/components/list/list-cell/list-cell.component.ts (hunk 7 above)
Comment (SilverGull) [reply src/kirby/components/list/list-cell/list-cell.component.ts]: didn't know that one. It is now fixed :-)

File: src/app/examples/examples.common.ts (hunk 9 above)
Comment (SilverGull) [reply src/app/examples/examples.common.ts]: I think it is 2. Now fixed. I will notify @askarby

File: src/kirby/components/avatar/avatar.component.tns.ts (hunk 1 above)
Comment (askarby) [reply src/kirby/components/avatar/avatar.component.tns.ts]: It's done to support layouts non @Input properties, in this usecase to use AvatarComponent with the row- and col-attributes used to control layout with GridLayout.

File: src/kirby/components/list/list-cell-line/list-cell-line.component.tns.ts (hunk 6 above)
Comment (askarby) [reply src/kirby/components/list/list-cell-line/list-cell-line.component.tns.ts]: Yes, unfortunately.... due to the "registerElement"-line having a strong dependency upon nativescript.

File: tsconfig.json (hunk 8 above)
Comment (askarby) [reply tsconfig.json]: It may have been, we'll roll this change back?

Comment (SilverGull) [reply tsconfig.json]: It is required until https://github.com/shripalsoni04/nativescript-webview-interface/pull/44 has been merged into master

Comment (jonasjensen-brolstark) [reply tsconfig.json]: I think we should roll this back 😄

**Pull request merged**